manifests/record/ns.pp

```diff
@@ -14,11 +14,19 @@
   validate_string($data)
   validate_string($host)
 
-  if !is_domain_name($zone) or $zone =~ /^[0-9\.]+$/ {
+  # Check corrected ns zone
+  $check_parameter_reverse = getparam(Dns::Zone[$zone], 'reverse')
+  if $check_parameter_reverse != 'reverse' and $check_parameter_reverse != true and !is_domain_name($zone) {
     fail("Define[dns::record::ns]: NS zone ${zone} must be a valid domain name.")
   }
+  if $check_parameter_reverse != 'reverse' and $check_parameter_reverse != true and $zone =~ /^[0-9\.]+$/ {
+    fail("Define[dns::record::ns]: NS zone ${zone} must be a valid domain name.")
+  }
+  if ( $check_parameter_reverse == 'reverse' or $check_parameter_reverse == true ) and $zone !~ /^((25[0-5]\.|2[0-4][0-9]\.|[01]?[0-9][0-9]?\.){2}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?){1}|(25[0-5]\.|2[0-4][0-9]\.|[01]?[0-9][0-9]?\.){1}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?){1}|(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?){1})$/ {
```

Comment:
This regex can be simplified to ... `"${zone}." !~ /^((25[0-5]\.|2[0-4][0-9]\.|[01]?[0-9][0-9]?\.){1,3})$?` - by adding a `.` to the end of `${zone}`, you don't need all the special cases; you just need to check for 1 to 3 of a valid octet followed by `.`